buildSrc/src/main/groovy/org/elasticsearch/gradle/vagrant/VagrantTestPlugin.groovy

```diff
@@ -452,6 +454,41 @@ class VagrantTestPlugin implements Plugin<Project> {
                 command BATS_TEST_COMMAND
             }
             packagingTest.dependsOn(packaging)
+
+            String task = "vagrant${boxTask}#repositoryTest"
+
+            String expectedRepoVersion = project.getProperties().get('repo.version')
+            String url = project.getProperties().get('repo.url')
+
+            Task createUrlFileTask = project.tasks.create("vagrant${boxTask}#createUrlFileTask", FileContentsTask) {
+                file "${batsDir}/archives/repo_url"
+                contents url
+            }
+
+            Task createRepoVersionFileTask = project.tasks.create("vagrant${boxTask}#createRepoVersionFileTask", FileContentsTask) {
+                file "${batsDir}/archives/repo_expected_version"
+                contents expectedRepoVersion
+            }
+
+            Task repotest = project.tasks.create(task, BatsOverVagrantTask) {
+                boxName box
+                environmentVars vagrantEnvVars
+                dependsOn up
+                finalizedBy halt
+                command "cd \$BATS_ARCHIVES && sudo bats --tap \$BATS_TESTS/99_upgrade_repos.$BATS"
+                doFirst {
+                    if (!expectedRepoVersion || !url) {
+                        throw new IllegalArgumentException("Task $task requires repo.version and repo.url properties")
+                    }
+                }
+            }
+
+            repotest.dependsOn 'vagrantSetUp', createUrlFileTask, createRepoVersionFileTask
         }
+
+        // one single repo test task to test all configured boxes
+        Task repotest = project.tasks.create('vagrantRepositoryTest')
+        List<String> boxes = listVagrantBoxes(project)
+        repotest.dependsOn boxes.collect { box -> 'vagrant' + box.capitalize().replace('-', '') + '#repositoryTest' }
```

Comment:
If you make `repotest` above the loop you can add the dependency on the repositoryTask in the loop which seems worth it.